### PR TITLE
[SOS]Gather openstack_edpm

### PR DESCRIPTION
**Pull request by gibizer**
openstack_edpm is added in https://github.com/sosreport/sos/pull/3414

JIRA [OSPRH-5744](https://issues.redhat.com/browse/OSPRH-5744)

### CODE REVIEW

**Comment by bogdando** [comment]
> /hold we don't a sosreport version with openstack_edpm released

any updates for this?

**Comment by gibizer** [comment]
> > /hold we don't a sosreport version with openstack_edpm released
> 
> any updates for this?

No update on my side. Currently I have no bandwidth to look at this.

**Comment by bogdando** [comment]
> we don't a sosreport version with openstack_edpm released

/unhold https://cbs.centos.org/koji/taskinfo?taskID=3768636

**Comment by bogdando** [comment]
could you please merge

**Comment by bogdando** [comment]
ok I'm double checkinig for https://cbs.centos.org/koji/taskinfo?taskID=3768636 contents...

File: collection-scripts/gather_sos [hunk 1]
```diff
@@ -8,7 +8,7 @@
 #   SOS_ONLY_PLUGINS: list of sos report plugins to use. Empty string to run
 #                     them all. Defaults to: block,cifs,crio,devicemapper,
 #                     devices,iscsi,lvm2,memory,multipath,nfs,nis,nvme,podman,
-#                     process,processor,selinux,scsi,udev
+#                     process,processor,selinux,scsi,udev,openstack_edpm
```
Comment (bogdando) [inline collection-scripts/gather_sos:11]: /hold sorry, I don't think I know how to check if this won't break today versions of must gather and sos reports that openstack-k8s-operators and this repo uses in CI

Comment (bogdando) [reply collection-scripts/gather_sos:11]: lets add the logs plugin to also have journal messages collected

File: collection-scripts/gather_sos [hunk 2]
```diff
@@ -38,7 +38,7 @@ else
 fi
 
 # Default to some plugins if SOS_ONLY_PLUGINS is not set
-SOS_ONLY_PLUGINS="${SOS_ONLY_PLUGINS-block,cifs,crio,devicemapper,devices,iscsi,lvm2,memory,multipath,nfs,nis,nvme,podman,process,processor,selinux,scsi,udev}"
+SOS_ONLY_PLUGINS="${SOS_ONLY_PLUGINS-block,cifs,crio,devicemapper,devices,iscsi,lvm2,memory,multipath,nfs,nis,nvme,podman,process,processor,selinux,scsi,udev,openstack_edpm}"
```
Comment (SeanMooney) [inline collection-scripts/gather_sos:41]: the plugins have not been updated to account for the fact that we are now logging to the journal

as a result we should consider if we need to enable the generic logs plugin by default to collect the combined journal log or if the podman plugin will collect the logs form all conatienrs.

Comment (SeanMooney) [reply collection-scripts/gather_sos:41]: this is the logs plugin 
https://github.com/sosreport/sos/blob/main/sos/report/plugins/logs.py
the alternative is to use the podman plugin with the logs option

https://github.com/sosreport/sos/blob/main/sos/report/plugins/podman.py#L118-L121

however that will not collect the libvirt logs.

we could do both but either option wont filter passwords or other info that is logged so we have to keep that in mind.

**Comment by gibizer** [comment]
/unhold 
based on https://github.com/openstack-k8s-operators/openstack-must-gather/pull/33/#issuecomment-2012397934

File: collection-scripts/gather_sos (hunk 1 above)
Comment (bogdando) [reply collection-scripts/gather_sos:11]: we have tested this in the linked ci-framework tooling, it works.
the docs follup is https://github.com/openstack-k8s-operators/openstack-must-gather/pull/18

we can merge this

File: collection-scripts/gather_sos (hunk 2 above)
Comment (bogdando) [reply collection-scripts/gather_sos:41]: we can do that in follow up.
Having the basic logging (journal etc. on edpm) is covered, as testing shows

**Comment by bogdando** [comment]
docs follow up https://github.com/openstack-k8s-operators/openstack-must-gather/pull/18

PTAL, good to go

**Comment by bogdando** [comment]
> /unhold based on #33

yes, "sos PR should appear in already released sos-4.6.1 downstream RPM (el8 and el9)"

**Comment by fmount** [comment]
> > /unhold based on #33
> 
> yes, "sos PR should appear in already released sos-4.6.1 downstream RPM (el8 and el9)"

I can approve this patch, it's actually good to finally start using the `openstack_edpm` profile. While I know that downstream we're already in `4.6.1`, I'm not sure about the status of upstream `CentOS`. If `openstack_edpm` is not found, sos will fail to start I suppose.
We have two workarounds for that:

1. install `epel-release` and update sos
2. use a `copr` repository like [1] to update the package during the edpm bootstrap

Of course this is something that should go in ci-framework.
Approving the patch and move the focus to [PR#1315](https://github.com/openstack-k8s-operators/ci-framework/pull/1315)
/cc @pablintino @rebtoor

[1] https://copr.fedorainfracloud.org/coprs/fmount/sos/

**Review by fmount — APPROVED** [review]
/lgtm

**Comment by SeanMooney** [comment]
> > > /unhold based on #33
> > 
> > 
> > yes, "sos PR should appear in already released sos-4.6.1 downstream RPM (el8 and el9)"
> 
> I can approve this patch, it's actually good to finally start using the `openstack_edpm` profile. While I know that downstream we're already in `4.6.1`, I'm not sure about the status of upstream `CentOS`. If `openstack_edpm` is not found, sos will fail to start I suppose. We have two workarounds for that:
> 
>     1. install `epel-release` and update sos
> 
>     2. use a `copr` repository like [1] to update the package during the edpm bootstrap
> 
> 
> Of course this is something that should go in ci-framework. Approving the patch and move the focus to [PR#1315](https://github.com/openstack-k8s-operators/ci-framework/pull/1315) /cc @pablintino @rebtoor
> 
> [1] https://copr.fedorainfracloud.org/coprs/fmount/sos/

ill point out that if the required package is not already release in centos and its part fo rhel then the rhel team have not followed our packaging process. all features delivered in a rhel release should be released in centos stream first. that would include functionality isn sos-reports.

if its not a part of core rhel and its an appstream they should still be in centos stream first
but we possibly would need to package it via rdo for c9s if that was not already provided in c9s for some other reason.

if we find the c9s version is not up to data that is a rhel bug and we should file its as such as outside of a embargoed cve fix we should never release functionality in rhel that has not been released in a package in c9s or its ecosystem first.

**Comment by bogdando** [comment]
> if we find the c9s version is not up to data that is a rhel bug

I checked the sos version in cs9 stream container image, and it proposes
sos.noarch 4.6.1-1.el9 from baseos repo, so no follow ups needed

**Pull request merged**